app/src/main/java/de/danoeh/antennapod/fragment/swipeactions/SwipeActions.java

```diff
@@ -0,0 +1,169 @@
+package de.danoeh.antennapod.fragment.swipeactions;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.graphics.Canvas;
+
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.ItemTouchHelper;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.Arrays;
+import java.util.List;
+
+import de.danoeh.antennapod.dialog.SwipeActionsDialog;
+import de.danoeh.antennapod.fragment.EpisodesFragment;
+import de.danoeh.antennapod.model.feed.FeedItem;
+import de.danoeh.antennapod.model.feed.FeedItemFilter;
+import de.danoeh.antennapod.view.viewholder.EpisodeItemViewHolder;
+import it.xabaras.android.recyclerview.swipedecorator.RecyclerViewSwipeDecorator;
+
+public class SwipeActions {
+    public static final String PREF_NAME = "SwipeActionsPrefs";
+    public static final String PREF_SWIPEACTION_RIGHT = "PrefsSwipeActionRight";
+    public static final String PREF_SWIPEACTION_LEFT = "PrefsSwipeActionLeft";
+
+    public static final int ADD_TO_QUEUE = 0;
+    public static final int MARK_UNPLAYED = 1;
+    public static final int START_DOWNLOAD = 2;
+    public static final int MARK_FAV = 3;
+    public static final int MARK_PLAYED = 4;
+    //indexes of swipeActions
+    public static final List<SwipeAction> swipeActions = Arrays.asList(new AddToQueue(),
+            new MarkUnplayed(), new StartDownload(),
+            new MarkFavourite(), new MarkPlayed());
+
+    RecyclerView recyclerView;
+    ItemTouchHelper itemTouchHelper;
+    Fragment fragment;
+    String tag;
+    FeedItemFilter filter = null;
+
+    public SwipeActions(Fragment fragment, String tag) {
+        this.fragment = fragment;
+        this.tag = tag;
+        itemTouchHelper();
+    }
+
+    public void setFilter (FeedItemFilter filter) {
+        this.filter = filter;
+    }
+
+    public SwipeActions attachTo(RecyclerView recyclerView) {
+        this.recyclerView = recyclerView;
+        itemTouchHelper.attachToRecyclerView(recyclerView);
+        return this;
+    }
+
+    private void itemTouchHelper() {
+        ItemTouchHelper.SimpleCallback simpleCallback = new ItemTouchHelper.SimpleCallback(0,
+                ItemTouchHelper.RIGHT | ItemTouchHelper.LEFT) {
```

Comment:
This prevents using a custom `ItemTouchHelper` in fragments (queue, for example, needs one). Instead, I think that this should be a `ItemTouchHelper.SimpleCallback` child class that can then be extended in the fragments if necessary.

Reply:
I'm not sure what the purpose of a "custom" ItemTouchHelper would be, but I could make the SimpleCallback separate and static I guess..

Reply:
Not static, please make it a class that can be extended. In the queue, we would then extend that class and add drag+drop in addition to swiping.

Reply:
I would keep that separate, the SimpleCallback already is so easy, I see no real need to extend it further, the queue doesn't need SwipeActions does it? It has only the swipe to remove function (which makes sense).
I would like to keep that wrapped in SwpiActions.java so it's a single line for each screen its added to (KISS&DRY).

Reply:
I'm pretty sure that once we add configurable swipe actions to one screen, we will get requests to add them to all other screens as well. So I think we should already plan with this in mind.

Reply:
Well i would add them to them search (just passing the TAG from the previous fragment (episodes or feed)
Queue I don't know, I can make it an inner class but by default I would like it to be wrapped in SwipeActions.

Reply:
ok i'll added SimpleSwipeCallback which i extened in QueueFragment, but SwipeActions can still be an almost one line in regular screens.

Reply:
how do i correctly pass a "fake" feeditemfilter from queuefragment?

Reply:
maybe you can quickly merge this <a href="https://github.com/AntennaPod/AntennaPod/pull/5231">PR</a> so i can do smth like `new FeedItemFilter(FeedItemFilter.QUEUED)`